src/Xenial.Tasty/Execution/TestRuntime/TestPipelineCompletedMiddleware.cs

```diff
@@ -0,0 +1,22 @@
+﻿
+namespace Xenial.Delicious.Execution.TestRuntime
+{
+    public static class TestPipelineCompletedMiddleware
+    {
+        public static TestExecutor UseTestPipelineCompleted(this TestExecutor executor)
+            => executor.UseRuntime(async (context, next) =>
+            {
+                try
+                {
+                    await next();
+                }
+                finally
+                {
+                    if (context.Remote != null)
+                    {
+                        await context.Remote.SignalTestPipelineCompleted();
```

Comment:
Again, what to do if connection breaks?

Reply:
Probably we should add some kind of reconnection, but I think we should postpone that to the blazor connection